Refactor documentation and document S3 (#812)
epinio/epinio: docs/user/howtos/setup-external-s3.md

## How to setup external S3 storage

> When the user runs `epinio push`, one of the steps involved is to store the requested version of the code,

Reviewer: ```suggestion
One of the steps involved in running `epinio push` is storing the requested version of the code
```

to be used for this purpose. This document describes how to point Epinio to another S3 compatible storage and skip the Minio installation.

The [`epinio install` command](../references/cli/epinio_install.md), has the following optional parameters:

```
--s3-endpoint
--s3-access-key-id
--s3-secret-access-key
--s3-bucket
--s3-location
--s3-use-ssl
```

To configure Epinio to store application sources to an external S3 compatible storage, at least the following should be set:

```
--s3-endpoint
--s3-access-key-id
--s3-secret-access-key
--s3-bucket
```

the other 2 are optional:

```
--s3-location
--s3-use-ssl
```

(Some implementations don't need the location (e.g. Minio) and `s3-use-ssl` has a default value of "false")

An example that points Epinio to AWS S3 will look like this:

```
epinio install \
--s3-endpoint s3.amazonaws.com \
--s3-access-key-id your_access_key_here \
--s3-secret-access-key your_secret_key_here \
--s3-bucket epinio_sources \
--s3-location eu-central-1
--s3-use-ssl
```

If the bucket doesn't exist, Epinio will try to create it when it first tries
to write to it. Make sure the access key and the access secret have enough permissions

> to create a bucket and the write to it.

Reviewer: ```suggestion
to create a bucket and then write to it.
```
